app/code/community/Algolia/Algoliasearch/Helper/Data.php

```diff
@@ -437,6 +437,46 @@ public function rebuildStoreCategoryIndexPage($storeId, $collectionDefault, $pag
         }
     }
 
+    /**
+     * Check if product can be index on Algolia
+     *
+     * @param Mage_Catalog_Model_Product $product
+     * @param int     $storeId
+     *
+     * @return bool
+     *
+     */
+    public function canProductBeReindexed(Mage_Catalog_Model_Product $product, $storeId)
+    {
+        if ($product->isDeleted() === true) {
+            throw (new Algolia_Algoliasearch_Model_Exception_ProductDeletedException())
+                ->withProduct($product)
+                ->withStoreId($storeId);
+        }
+
+        if ($product->getStatus() == Mage_Catalog_Model_Product_Status::STATUS_DISABLED) {
```

Comment:
Can a strict (`===`) comparison be used?

Reply:
Strict typing threw issues on travis interestingly enough. $product->getStatus() will not guarantee a int return to match strict typing.

Reply:
Magento /o\ OK :))